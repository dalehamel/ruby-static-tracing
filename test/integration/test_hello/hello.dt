global*:::hello_test
{
  printf("Attaching 1 probe...\n%s\n", copyinstr(arg0))
}
