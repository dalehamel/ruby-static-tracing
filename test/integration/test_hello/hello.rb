# frozen_string_literal: true

require 'ruby-static-tracing'
STDOUT.sync = true

t = StaticTracing::Tracepoint.new('global', 'hello_test', String)
p = StaticTracing::Provider.fetch(t.provider)
puts p.enable

Signal.trap('USR2') do
  puts "TRAP #{t.enabled?}"
  t.fire('Hello world') if t.enabled?
  sleep 5
end

loop { puts t.enabled?; sleep 1 }
