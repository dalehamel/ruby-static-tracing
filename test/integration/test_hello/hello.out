Attaching 1 probe...
Hello world
